### ISSUE TITLE
Undocumented behavior for jsonBody[None]

### ISSUE THREAD

**Issue by Awethon** [softwaremill/tapir#3043]
Circe itself won't parse empty string as None
The logic that makes it possible is contained somewhere else (where by the way?) and is not documented.

Should I expect `jsonBody[Option[T]]` to work in the same way for all servers/clients for both input and output?

**Comment by adamw** [comment]
Yes, this is captured by the `IsOption` typeclass, required here: https://github.com/softwaremill/sttp/blob/e7ed7720a3ae4389c4b755b476e94bafb26ba694/json/circe/src/main/scala/sttp/client4/circe/SttpCirceApi.scala#L23

This is used in `JsonInput.sanitize`: 
https://github.com/softwaremill/sttp/blob/3f1cb61c0297f15495fc2058c5ecf41391158e6e/json/common/src/main/scala/sttp/client4/JsonInput.scala#L4-L6

The same logic is used for all json integrations.

This indeed might not be documented anywhere (well it is now, in this issue ;) ), but I think it matches the intuition - the parsing only happens if there is a body. Or rather, if the body is empty, then you get a `None`.

**Comment by adamw** [comment]
Sorry, I responsed from sttp-client's point of view, not tapirs :)

In case of tapir, the behavior is similar, if the body is empty, we try to decode `null`, as implemented here: https://github.com/softwaremill/tapir/blob/7a9e8872a7e3faaf2f7d94bb7e4ab01beb72c80b/core/src/main/scala/sttp/tapir/Codec.scala#L568

**Comment by adamw** [comment]
@Awethon where would you look for docs on this particular feature? :)

**Comment by adamw** [comment]
Here you go: https://github.com/softwaremill/tapir/commit/f5a4217e0309936ff926b076409d2768c5038414 :)
